unittests/integration/measure_reset_tester.cpp

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "CUDAQTestUtils.h"
+
+#include <cudaq.h>
+#include <iostream>
+
+TEST(MeasureResetTester, checkBug981) {
+
+  auto bar = []() __qpu__ {
+    cudaq::qubit a;
+    cudaq::x(a);
+    [[maybe_unused]] auto a0 = cudaq::mz(a);
+    cudaq::reset(a);
+    [[maybe_unused]] auto a1 = cudaq::mz(a);
+  };
+
+  std::cout << "Bar:\n";
+  auto result = cudaq::sample(bar);
```

Comment:
Slight optimization since this test requires repeated state vector evolutions and since there is no expected variability in per-shot answers.
```suggestion
  auto result = cudaq::sample(/*shots=*/10, bar);
```